doc/man3/PKCS12_create.pod

```diff
@@ -34,6 +38,10 @@ I<keytype> is the type of key.
 PKCS12_create_ex() is identical to PKCS12_create() but allows for a library context
 I<ctx> and property query I<propq> to be used to select algorithm implementations.
 
+PKCS12_create_ex2() is identical to PKCS12_create_ex() but allows for a user defined
+callback I<cb> of type B<PKCS12_create_cb> to be specified and also allows for an 
+optional argument I<cbarg> to be passed back to the callback.
```

Comment:
We should include the `PKCS12_create_cb` typedef in the man page itself. E.g see https://www.openssl.org/docs/man3.0/man3/EVP_PKEY_CTX_set_cb.html for an example. Also we should say a little more about the callback, i.e. that it is called for every safebag added to the PKCS12 structure, and what the return values from the callback mean.

Reply:
I'll look at those existing tests and see how I can expand them for this PR.

Reply:
The test/pkcs12_api_test.c doesn't exactly look like it's got many tests, esp given how big/complex the PKCS12 API is

Reply:
@mattcaswell am I missing something but a grep of the existing PKCS12_create and PKCS12_create_ex under the test subdirectory doesn't show anything

Reply:
```
export OPENSSL_CONF=test/jdk_trusted.cnf
/usr/local/openssl/bin/openssl pkcs12 -export -out out7.p12 -in test/certs/ee-cert.pem -nokeys -passout pass: -certpbe NONE
/usr/local/openssl/bin/openssl pkcs12 -info -in out7.p12 -passin pass:
MAC: sha256, Iteration 2048
MAC length: 32, salt length: 8
PKCS7 Data
Certificate bag
Bag Attributes
    Trusted key usage (Oracle): Any Extended Key Usage (2.5.29.37.0)
…
…

```

Reply:
meh never mind, my regex in the grep was wrong...I needed to escape the () doh

Reply:
> The test/pkcs12_api_test.c doesn't exactly look like it's got many tests, esp given how big/complex the PKCS12 API is

and

> @mattcaswell am I missing something but a grep of the existing PKCS12_create and PKCS12_create_ex under the test subdirectory doesn't show anything

There are some additional tests in pkcs12_format_test.c, one of which calls `PKCS12_create`. But, generally, no you're not missing anything. PKCS12 is under tested. These days it is our policy that all new functionality needs to have tests associated with it (see our [testing policy](https://www.openssl.org/policies/technical/testing.html)). But this was not always the case. When the code was originally added (from git logs it seems to have been there since 1999), there was no requirement to add tests. We mostly add new tests when we are making changes, but the PKCS12 code has seen relatively little change compared to other parts of OpenSSL and so coverage there is insufficient.